cmd/network/aws.go

```diff
@@ -90,42 +80,20 @@ func (e *EgressVerification) setupForAws(ctx context.Context) (*aws.Config, erro
 }
 
 // generateAWSValidateEgressInput is an opinionated interface in front of osd-network-verifier.
-// Its input is an OCM internal/external ClusterId and it returns the corresponding input to osd-network-verifier with
+// Its input is an OCM internal/external ClusterId, and it returns the corresponding input to osd-network-verifier with
 // default AWS tags, one of the cluster's private subnet IDs, and the cluster's master security group.
 // Can override SecurityGroupId and SubnetIds.
-func (e *EgressVerification) generateAWSValidateEgressInput(ctx context.Context, region string) ([]*onv.ValidateEgressInput, error) {
-	// We can auto-detect information from OCM
-	if e.cluster != nil {
-		if e.cluster.Product().ID() != "rosa" &&
-			e.cluster.Product().ID() != "osd" &&
-			e.cluster.Product().ID() != "osdtrial" {
-			return nil, fmt.Errorf("only supports rosa, osd, and osdtrial, got %s", e.cluster.Product().ID())
-		}
-	}
-
-	input, err := defaultValidateEgressInput(ctx, e.cluster, region)
+func (e *EgressVerification) generateAWSValidateEgressInput(ctx context.Context, platform cloud.Platform) ([]*verifier.ValidateEgressInput, error) {
```

Comment:
A lot of the changes here are just moving common bits of the setup that doesn't differ between clouds into shared functions